enso/experiment/rationalized.py

```diff
@@ -264,21 +266,71 @@ def _featurize(self, doc):
 
 
 @Registry.register_experiment(ModeKeys.RATIONALIZED, requirements=[("Featurizer", "PlainTextFeaturizer")])
-class RationaleInformedLRCV(BaseRationaleGridSearch):
+class DistReweightedGloveByClassClassifierCV(BaseRationaleGridSearch):
+    """
+    Weights words by cosine similarity to the mean of the rationale vector representations per class
+
+    """
+    def _train_rationale_model(self, docs, rationale_docs, labels=None):
+        rationale_vecs_by_class = defaultdict(list)
+        for doc, label in zip(rationale_docs, labels):
+            if doc and doc.has_vector and np.any(np.nonzero(doc.vector)):
+                rationale_vecs_by_class[label].append(
+                    doc.vector / np.linalg.norm(doc.vector)
+                )
+        rationale_proto_by_class = {
+            label: np.mean(rationale_vecs, axis=0)
+            for label, rationale_vecs in rationale_vecs_by_class.items()
+        }
+        self.normalized_rationale_proto_by_class = OrderedDict({
+            label: rationale_proto / np.linalg.norm(rationale_proto)
+            for label, rationale_proto in rationale_proto_by_class.items()
+        })
 
-    def _train_rationale_model(self, docs, rationale_docs):
+    def _rationale_weight(self, word, rationale_proto):
+        cosine_sim = np.dot(word.vector / np.linalg.norm(word.vector), rationale_proto)
```

Comment:
By using cosine similarity we give a strong negative weight to anything that differs a lot, would a cosine distance not make more sense.

Reply:
that's a reasonable hypothesis for sure! However, I tested and it seems to perform less well (closer to the baseline). My intuition is that the negative weights end up creating even more of an emphasis on the rationale vector direction.